marathon/marathon.go

```diff
@@ -106,6 +115,51 @@ func (m Marathon) Leader() (string, error) {
 	return leaderResponse.Leader, err
 }
 
+func (m Marathon) EventStream() (*Streamer, error) {
+	subURL := m.url("/v2/events")
+	// url lib doesnt support multiple params with the same name, last one is used
+	// TODO(tz) - encode params at least
+	subURL += fmt.Sprintf("?%s", m.subscribeFilterParams())
+
+	err := m.leaderPoll()
+	if err != nil {
+		log.WithError(err).Fatal("Leader poll failed. Check marathon and previous errors. Exiting")
+	}
+
+	return &Streamer{
+		subURL:   subURL,
+		marathon: m,
+		client: &http.Client{
+			Transport: m.client.Transport},
+	}, nil
+}
+
+// leaderPoll just blocks until configured myleader is equal to
+// leader returned from marathon (/v2/leader endpoint)
+func (m Marathon) leaderPoll() error {
```

Comment:
Instead of pool I think we can use SSE to keep connection. When it breaks we will know leader has changed and we need to retry. This could be done in separate PR.